common/output.h

```diff
@@ -39,11 +39,26 @@ int mdprintf(int desc, const char *str, ...) __attribute__((format(printf, 2, 3)
 int mdprintf(int desc, const char *str, ...);
 #endif
 
-#ifdef __GNUC__
-int logg(const char *str, ...) __attribute__((format(printf, 1, 2)));
-#else
-int logg(const char *str, ...);
-#endif
+/*
+ * legend:
+ * NAME     OLD        EXPLAIN
+ * INFO     none or ~   normal
+ * INFO_NF  #           normal, no foreground
+ * DEBUG    *           debug, verbose
+ * DEBUG_NV $           debug, non-verbose
+ * WARNING  ^           warning
+ * ERROR    !           ERROR
+ */
+enum loglevel_t{
+    INFO, 
+    INFO_NF, 
+    DEBUG, 
+    DEBUG_NV, 
+    WARNING, 
+    ERROR
+};
```

Comment:
The CI failed on Windows because of a namespace collision with "ERROR" being defined in the Windows headers.  I'm a little surprised actually that we didn't run into that with any of the others. 

I guess we should prefix them with something recognizable, like "`LOGG_`".

It would also be nice to make this a typedef so we can reference the `loglevel_t` enum without having to use the worth `enum`.  I'm thinking:

```c
typedef enum loglevel {
    LOGG_INFO,
    LOGG_INFO_NF,
    LOGG_DEBUG,
    LOGG_DEBUG_NV,
    LOGG_WARNING,
    LOGG_ERROR
} loglevel_t;
```

Sorry about the hassle renaming everything, again.

Reply:
Well, it's not that hard job to rename nowadays with vs-code. Changed the enum to a typedef